web/client/map/hooks/use-map-tool.js

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2020, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+*/
+
+import {useEffect, useRef, useState} from "react";
+
+const useMapTool = (mapType, tool) => {
```

Comment:
We should document this component, in order allow a reuse. But before please notice this:
In my idea `import("../" + mapType + "/" + tool)` should be externalized, because it forces all the supports to stay in the same directory in the product. The purpose of this work is also to provide a reusable model to implement our own supports, and we should not be forced to have them inside maptore.

@allyoucanmap @MV88 and @mbarto Please tell me what you think about this?

I propose to pass to the hook the `load` promise as parameter, instead of `mapType` and `tool` but I don't know if it is possible with hooks and lifecycle stuff.